specifyweb/frontend/js_src/lib/components/DataModel/resource.ts

```diff
@@ -225,12 +225,37 @@ export function resourceOn(
 export const parseJavaClassName = (className: string): string =>
   className.split('.').at(-1) ?? '';
 
+const interactionTablesWithPreps = [
+  'Disposal',
+  'Loan',
+  'Gift',
+  'ExchangeIn',
+  'ExchangeOut',
+  'Borrow',
+] as const;
+type InteractionTable = typeof interactionTablesWithPreps[number];
+const interactionTablesPrepsFieldName: RR<InteractionTable, string> = {
+  Disposal: 'disposalPreparations',
+  Loan: 'loanPreparations',
+  Gift: 'giftPreparations',
+  ExchangeOut: 'exchangeOutPreps',
+  ExchangeIn: 'exchangeInPreps',
+  Borrow: 'borrowPreparations',
+};
 export function getFieldsToNotClone(
   table: SpecifyTable,
   cloneAll: boolean
 ): RA<string> {
-  const fieldsToClone = getCarryOverPreference(table, cloneAll);
+  let fieldsToClone = getCarryOverPreference(table, cloneAll);
   const uniqueFields = getUniqueFields(table);
+  if (
+    interactionTablesWithPreps.includes(table.name as InteractionTable) &&
+    cloneAll
```

Comment:
are you sure this logic should apply only in the `cloneAll` cases?
it seems like it should apply in `!cloneAll` case too...